api/urls.py

```diff
@@ -15,48 +22,97 @@ def register_route(prefix, viewset):
-# register_creative_work_route(r'dataset')
-# register_creative_work_route(r'dissertation')
-# register_creative_work_route(r'lesson')
-# register_creative_work_route(r'poster')
-# register_creative_work_route(r'preprint')
-# register_creative_work_route(r'presentation')
-# register_creative_work_route(r'project')
-# register_creative_work_route(r'projectregistration')
-# register_creative_work_route(r'report')
-# register_creative_work_route(r'section')
-# register_creative_work_route(r'software')
-# register_creative_work_route(r'thesis')
-# register_creative_work_route(r'workingpaper')
-
-# # creative work attributes
-# register_route(r'tag', views.TagViewSet)
-# register_route(r'subject', views.SubjectViewSet)
-# register_route(r'extra', views.ExtraDataViewSet)
-
-# # agent routes
-# register_route(r'person', views.PersonViewSet)
-# register_route(r'organization', views.OrganizationViewSet)
-# register_route(r'institution', views.InstitutionViewSet)
-
-# # identifier routes
-# register_route(r'agentidentifier', views.AgentIdentifierViewSet)
-# register_route(r'workidentifier', views.WorkIdentifierViewSet)
-
-# # relation routes
-# register_route(r'agentrelation', views.AgentRelationViewSet)
-# register_route(r'workrelation', views.WorkRelationViewSet)
-# register_route(r'contribution', views.ContributionViewSet)
-# register_route(r'award', views.AwardViewSet)
+class BaseShareSerializer(serializers.ModelSerializer):
+
+    def __init__(self, *args, **kwargs):
+        # super hates my additional kwargs
+        sparse = kwargs.pop('sparse', False)
+        version_serializer = kwargs.pop('version_serializer', False)
+        super(BaseShareSerializer, self).__init__(*args, **kwargs)
+
+        if sparse:
+            # clear the fields if they asked for sparse
+            self.fields.clear()
+        else:
+            # remove hidden fields
+            excluded_fields = ['change', 'uuid', 'sources']
+            for field_name in tuple(self.fields.keys()):
+                if 'version' in field_name or field_name in excluded_fields:
+                    self.fields.pop(field_name)
+
+            if not version_serializer:
+                # add links to related objects
+                self.fields.update({
+                    'links': fields.LinksField(links=self.Meta.links, source='*')
+                })
+
+        # version specific fields
+        if version_serializer:
+            self.fields.update({
+                'action': serializers.CharField(max_length=10),
+                'persistent_id': serializers.IntegerField()
+            })
+
+        # add fields with improper names
+        self.fields.update({
+            'type': fields.TypeField(),
+        })
+
+    class Meta:
+        links = ('versions', 'changes', 'rawdata')
+
+    # http://stackoverflow.com/questions/27015931/remove-null-fields-from-django-rest-framework-response
+    def to_representation(self, instance):
+        def not_none(value):
```

Comment:
This function serves no purpose. `lambda x: x[1] is not None` is much easier to follow.